### ISSUE TITLE
Document: When is mandatory define each value for SqlConfig.TransactionMode for @SqlConfig?

### ISSUE THREAD

**Issue by manueljordan** [spring-projects/spring-framework#30706]
Hello Spring Team

The question is available on SO at:

* [When is mandatory define each value for SqlConfig.TransactionMode for @SqlConfig?](https://stackoverflow.com/questions/76382634/when-is-mandatory-define-each-value-for-sqlconfig-transactionmode-for-sqlconfig)

But the current and unique "answer" is not neither valid nor _realistic_ (AI is involved)

Therefore more details for the [SqlConfig.TransactionMode](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/test/context/jdbc/SqlConfig.TransactionMode.html) enum would valuable to understand when choice each value explicitly.

Thanks for your understanding.

**Comment by snicoll** [comment]
I agree that there are a lot of information and rules about how this behaves. The reference guide could perhaps use some more examples.

cc @sbrannen
